openprescribing/frontend/utils/bnf_hierarchy.py

```diff
@@ -6,20 +6,25 @@ def simplify_bnf_codes(bnf_codes):
 
         1. every BNF code that belongs to one of these prefixes is in the original list,
         2. every code in the original list belongs to exactly one prefix,
-        3. no prefix is a prefix of another prefix (I think this follows from 2).
+        3. no prefix is a prefix of another prefix (I think this follows from 2),
+        4. every prefix is present in the prescribing data.
 
     A BNF prefix may actually be a full BNF code.
     """
 
     all_bnf_codes = get_all_bnf_codes()
+
+    # Drop any BNF codes for which we don't have prescribing.
+    bnf_codes = set(bnf_codes) & all_bnf_codes
+
     return _prune_paths(bnf_codes, all_bnf_codes)
 
 
 def get_all_bnf_codes():
```

Comment:
Should this be cached?

Reply:
No, it's only called by the pipeline and anyway it's pretty quick.